### ISSUE TITLE
ISBN Generator output does not follow convention for author names

### ISSUE THREAD

**Issue by duelle** [JabRef/jabref#2343]
JabRef 3.7
Archlinux Kernel 4.8.12-2
Java 1.8.0_112 (OpenJDK)

The results of the ISBN Generator are not following the conventions for author names which leads to references that cannot be resolved in LaTeX.

Steps to reproduce:
1. In menu: BibTeX -> New Entry
2. Change ID type to "ISBN"
3. Enter for ID: 0321331885
4. Press "Generate"
5. The author field content is not correctly formatted:
Result: 
```David Steinberg, Frank Budinsky, Marcelo Paternostro, Ed Merks```
Should be: 
```Steinberg, David and Budinsky, Frank and Paternostro, Marcelo and Merks, Ed```

The result of the request http://www.ebook.de/de/tools/isbn2bibtex?isbn=0321331885 is:
```
@book{9780321331885,
  Author = {David Steinberg, Frank Budinsky, Marcelo Paternostro, Ed Merks},
  Title = {EMF. Eclipse Modeling Framework},
  Publisher = {Addison Wesley},
  Year = {2009},
  Date = {2009-01-12},
  PageTotal = {XXIX},
  EAN = {9780321331885},
  ISBN = {0321331885},
  URL = {http://www.ebook.de/de/product/6903954/david_steinberg_frank_budinsky_marcelo_paternostro_ed_merks_emf_eclipse_modeling_framework.html}
}
```
The entry above is just an example. I tested this with several other books having mutliple authors with a similar outcome.
Obviously the JabRef implementation is not the issue but the service that is hardcoded into the application.

**Comment by stefan-kolb** [comment]
So this is a problem of the returned BibTeX of ebook.de?! The only thing we could do is to run a normalization on the returned data. This is one thing you should be able to do manually as well to fix this issue. 

![image](https://cloud.githubusercontent.com/assets/2141507/20969632/2c5ca426-bc8a-11e6-9bae-4468e0620ef6.png)

As you can see the BibTeX key is also not formated as expected. 

**For me, the question here is:**
@JabRef/developers Should we add a middleware after the fetchers to fix such issues (BibTeX key generation and other normalization) or should this be the responsibility of the user?

**Comment by koppor** [comment]
Responsibility of the fetcher. We have `net.sf.jabref.logic.importer.SearchBasedParserFetcher#doPostCleanup`

**Comment by duelle** [comment]
@stefan-kolb Ah, good to know, thank you for the hint!

If it is possible to detect that the format is not correct, one could offer this as a dialog after having generated the entry, because I did not know that this normalization feature even exists..

**Comment by tobiasdiez** [comment]
`doPostCleanup` should fix real problems while normalization etc. should be handled by the user (as some users want e.g. BibLaTeX, or maybe even keep the above name list...which by the way is not against the "bibtex standard" as far as I remember). Moreover, we already have two issues (#895 #1283) which propose to perform a user-defined cleanup after import/web search. Thus I vote for closing this issue as duplicate.

**Comment by koppor** [comment]
Unfortunately, the above name list is not following the bibtex standard. See http://tex.stackexchange.com/a/36397/9075. Thus it should be fixed in `doPostCleanup`. I will work together with @duelle to create a PR.

**Comment by koppor** [comment]
Should be fixed in JabRef 3.8 with #2374.
